### ISSUE TITLE
Code generator cannot generate proper Ruby codes for interfaces extending multiple interfaces

### ISSUE THREAD

**Issue by tomoasleep** [mtsmfm/language_server-protocol-ruby#28]
Some interfaces in LSP extends multiple interfaces such as the following `WorkspaceSymbolParams`.
However, in Ruby code, the interface class only extends the last interface class (`WorkspaceSymbolParams` class only extends `PartialResultParams` class), so the Ruby class is not compatible with the corresponding LSP interface.

For example, this `WorkspaceSymbolParams` interface ([link](https://microsoft.github.io/language-server-protocol/specifications/specification-current/#workspace_symbol)) extends `WorkDoneProgressParams` interface ([link](https://microsoft.github.io/language-server-protocol/specifications/specification-current/#workDoneProgress)) which has `workDoneToken` property and `PartialResultParams` interface ([link](https://microsoft.github.io/language-server-protocol/specifications/specification-current/#partialResultParams)) which has `partialResultToken` property.
Therefore, this `WorkspaceSymbolParams` interface has `query`, `workDoneToken`, and `partialResultToken` properties.

```ts
/**
 * The parameters of a Workspace Symbol Request.
 */
interface WorkspaceSymbolParams extends WorkDoneProgressParams,
	PartialResultParams {
	/**
	 * A query string to filter symbols by. Clients may send an empty
	 * string here to request all symbols.
	 */
	query: string;
}
```

However, in Ruby code, the interface class `WorkspaceSymbolParams` only extends `PartialResultParams` class, so `WorkspaceSymbolParams`  does not allow `workDoneToken` propety of `WorkDoneProgressParams` class.

https://github.com/mtsmfm/language_server-protocol-ruby/blob/e9a357f576d5bef81de192dd98e7ddfe098b06e3/lib/language_server/protocol/interface/workspace_symbol_params.rb#L1-L38

**Comment by mtsmfm** [comment]
Thank you for reporting, I'll have a look at

**Comment by mtsmfm** [comment]
@tomoasleep Fixed in 3.16.0.1. Thanks for your help!
